src/module/rolls/roll.ts

```diff
@@ -27,48 +52,81 @@ export type SourcedValue<T = number> = {
   value: T
 }
 
-// Input to rolling and resolution
+/**
+ * Input to rolling and resolution
+ */
 export interface PreRollOptions {
-  // Exactly one of `progress` or `action` is required
-
-  // indicates this is a progress roll, no action die will be rolled
+  // TODO: candidate for being a generic type with some variable props dependent on boolean "isProgressRoll"?
+  /**
+   * Exactly one of `progress` or `action` is required
+   */
+
+  /**
+   * indicates this is a progress roll, no action die will be rolled
+   */
   progress?: SourcedValue
 
-  // Indicates this is an action roll
+  /**
+   * Indicates this is an action roll
+   */
   stat?: SourcedValue
+  /**
+   * Indicates this is an action roll
+   */
   adds?: number
 
-  // Negative momentum can cancel an action die
+  /**
+   * Negative momentum can cancel an action die
+   */
   momentum?: number
 
   automaticOutcome?: SourcedValue<ROLL_OUTCOME>
-  // As in Armored #1
+  /**
+   * As in Armored #1
+   */
   presetActionDie?: SourcedValue
-  // As in Sleuth #1
+  /**
+   * As in Sleuth #1
+   */
   extraChallengeDice?: SourcedValue
 
   // Decided before the roll, but kept around for resolving updates later
-  moveId?: string // For custom moves
-  moveDfId?: string // for "official" moves
+
+  /**
+   * For custom moves
+   */
+  moveId?: string
+  /**
+   * for "official" moves
+   */
+  moveDfId?: string
   actorId?: string
 }
 
 // Input to rendering, can be updated after the fact
 export interface PostRollOptions {
-  // As in Kinetic #2
-  // if present and non-empty, will suggest in the chat-card UI
+  /**
+   * As in Kinetic #2. if present and non-empty, will suggest in the chat-card UI
+   */
   suggestedAdds?: SourcedValue[]
   adds?: SourcedValue[]
 
-  // As in Loyalist #3
+  /**
+   * As in Loyalist #3
+   */
   replacedChallenge1?: SourcedValue
   replacedChallenge2?: SourcedValue
 
-  // As in Brawler #2 or Take Decisive Action
+  /**
+   * As in Brawler #2 or Take Decisive Action
+   */
   allowReplacingOutcome?: boolean
   replacedOutcome?: SourcedValue<ROLL_OUTCOME>
 }
 
+/**
+ * This is intended as a way for (e.g.) a macro author to make use of this class without having to instantiate a complex object as an input. It's a shortcut, and we're not using it anywhere in the code. Given that use case, I don't think we need a TODO here to support passing in assets, because that's logic for a different layer — this class does rolling and storage, it doesn't know about assets.
```

Comment:
Probably doesn't need to be in the code.

Reply:
i'll make the tsdoc annotation more concise. main reason i dumped your description in there is so i'd remember that "IronswornRoll" has that specific usage. another possibility is renaming the class... but ideally it has both an informative name *and* a tsdoc annotation, IMO.